File: frontend/olympic_it_project/olympic_it_project/lib/dto/auth/login/login_response.dart
class LoginResponse {
  final String accessToken;
  final String refreshToken;
  final String roleName;
  final int userId;

  LoginResponse({
    required this.accessToken,
    required this.refreshToken,
    required this.roleName,
    required this.userId
  });

  factory LoginResponse.fromJson(Map<String, dynamic> json) {
    return LoginResponse(accessToken: json["accessToken"] ?? "",
    refreshToken: json["refreshToken"] ?? "",
    roleName: json["roleName"] ?? "",
    userId: json["userId"] ?? 0
    );
  }
}

File: admin-web/js/dto/auth/login_response.js
export class LoginResponse {
  constructor({ accessToken, refreshToken, userId, roleName }) {
    this.accessToken = accessToken;
    this.refreshToken = refreshToken;
    this.userId = userId;
    this.roleName = roleName;
  }

  static fromJson(json) {
    return new LoginResponse({
      accessToken: json.accessToken,
      refreshToken: json.refreshToken,
      userId: json.userId,
      roleName: json.roleName,
    });
  }
}